UI/src/javascript/search-results.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+import axios from "axios";
+
+const SearchResults = (() => {
+  const getAddShortlistTuitionPartnerFetchUrl = (seoUrl) =>
+    `/search-results?handler=AddShortlistedTuitionPartner&tuitionPartnerSeoUrl=${seoUrl}`;
+
+  const getRemoveShortlistTuitionPartnerFetchUrl = (seoUrl) =>
+    `/search-results?handler=RemoveShortlistedTuitionPartner&tuitionPartnerSeoUrl=${seoUrl}`;
+
+  const getSeoUrlFromCheckboxId = (id) => id && id.replace("shortlist-cb-", "");
+
+  const isFetchResponseDataValid = (response) =>
+    response &&
+    response.isCallSuccessful !== "undefined" &&
+    response.totalShortlistedTuitionPartners !== "undefined";
+
+  const shortlistCheckboxes = document.querySelectorAll(
```

Comment:
If the shortlist checkboxes aren't in the DOM when this script is loaded and are later added, then they won't be included when `init()` is called. Given that `shortlistCheckboxes` isn't used outside of `init()` then you could just move this `document.querySelectorAll` into `init()`.